src/main/java/reposense/authorship/FileInfoExtractor.java

```diff
@@ -167,13 +167,13 @@ private static void getAllFileInfo(RepoConfiguration config, Path directory, Lis
                 String relativePath = filePath.toString().substring(config.getRepoRoot().length());
                 if (Files.isDirectory(filePath)) {
                     getAllFileInfo(config, filePath, fileInfos);
-                }
-
-                if (Format.isInsideWhiteList(relativePath, config.getFormats())) {
-                    try {
-                        fileInfos.add(generateFileInfo(config.getRepoRoot(), relativePath));
-                    } catch (InvalidPathException ipe) {
-                        logger.warning(String.format(INVALID_FILE_PATH_MESSAGE_FORMAT, filePath));
+                } else {
+                    if (Format.isInsideWhiteList(relativePath, config.getFormats())) {
```

Comment:
Does it work if you use `continue` in 
```
if (Files.isDirectory(filePath)) {
    getAllFileInfo(config, filePath, fileInfos);	                    getAllFileInfo(config, filePath, fileInfos);
}
```
Otherwise, use `elseif`.

Also, let's have a test. You can add a branch that contains the required test files in `reposense/testrepo-Alpha` 😃

Reply:
@yong24s It works.

And also added a branch https://github.com/reposense/testrepo-Alpha/tree/directory-with-valid-whitelisted-name and it passes the test